catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStorePublisher.java

```diff
@@ -0,0 +1,214 @@
+            LOGGER.warn("Reference or BundleContext was null/unset.");
+            return;
+        }
+
+        RegistryStore registryStore = bundleContext.getService(reference);
+
+        Boolean previousState = registryStoreMap.put(registryStore.getConfigurationPid(),
+                registryStore.isAutoPush());
+
+        SourceMonitor storeRegisteredSourceMonitor = new SourceMonitor() {
+            @Override
+            public void setAvailable() {
+                if (registryStore.isAutoPush()) {
+                    registryPublish(registryStore, PUBLISH);
+                }
+            }
+
+            @Override
+            public void setUnavailable() {
+            }
+        };
+
+        if (registryStore.isAvailable(storeRegisteredSourceMonitor) && registryStore.isAutoPush()) {
+            if (previousState == null || !previousState) {
+                registryPublish(registryStore, PUBLISH);
+            }
+        }
+    }
+
+    public void unbindRegistryStore(ServiceReference<RegistryStore> reference) {
+
+        BundleContext bundleContext = getBundleContext();
+
+        if (reference != null && bundleContext != null) {
+            RegistryStore registryStore = bundleContext.getService(reference);
+
+            registryStoreMap.remove(registryStore.getConfigurationPid());
+        }
+    }
+
+    void registryPublish(RegistryStore registryStore, String publish) {
+
+        if (registryStore.getRegistryId()
+                .isEmpty()) {
+            LOGGER.warn(String.format("RegistryStore missing id. Unable to complete %s request.",
+                    publish));
+            return;
+        }
+
+        executor.schedule(() -> {
+            try {
+                Optional<Metacard> registryIdentityMetacardOpt =
+                        Security.runAsAdminWithException(() -> federationAdminService.getLocalRegistryIdentityMetacard());
+
+                if (registryIdentityMetacardOpt.isPresent()) {
+                    Metacard registryIdentityMetacard = registryIdentityMetacardOpt.get();
+                    String localRegistryId = registryIdentityMetacard.getAttribute(
+                            RegistryObjectMetacardType.REGISTRY_ID)
+                            .getValue()
+                            .toString();
+                    Security.runAsAdminWithException(() -> {
+                        if (publish.equals(PUBLISH)) {
+                            registryPublicationService.publish(localRegistryId,
+                                    registryStore.getRegistryId());
+                        } else {
+                            registryPublicationService.unpublish(localRegistryId,
+                                    registryStore.getRegistryId());
+                        }
+                        return null;
+                    });
+                }
+
+            } catch (Exception e) {
+                LOGGER.error(String.format("Failed to %s registry configuration to %s",
```

Comment:
Just a note here, you can do `LOGGER.error("Failed to {} registry configuration to {}", arg1, arg2)` instead of using String.format